src/ensembl/src/content/app/entity-viewer/gene-view/components/protein-domain-image/ProteinDomainImage.tsx

```diff
@@ -0,0 +1,160 @@
+import React from 'react';
+import classNames from 'classnames';
+import { scaleLinear, ScaleLinear } from 'd3';
+
+import { getFeatureCoordinates } from 'src/content/app/entity-viewer/shared/helpers/entity-helpers';
+import { Transcript } from 'src/content/app/entity-viewer/types/transcript';
+import { ProteinDomainsResources } from 'src/content/app/entity-viewer/types/product';
+
+import styles from './ProteinDomainImage.scss';
+
+const BLOCK_HEIGHT = 18;
+const TRACK_HEIGHT = 24;
+
+export type ProteinDomainImageProps = {
+  transcript: Transcript;
+  width: number; // available width for drawing, in pixels
+  classNames?: {
+    track?: string;
+    domain?: string;
+  };
+};
+
+type ProteinDomainImageData = {
+  [resource_type: string]: {
+    [resource_description: string]: {
+      start: number;
+      end: number;
+    }[];
+  };
+};
+
+export const getDomainsByResourceGroups = (
+  proteinDomainsResources: ProteinDomainsResources
+) => {
+  const proteinDomains: ProteinDomainImageData = {};
+
+  Object.keys(proteinDomainsResources).forEach((resourceName) => {
+    if (!proteinDomains[resourceName]) {
+      proteinDomains[resourceName] = {};
+    }
+
+    proteinDomainsResources[resourceName].domains.map((domain) => {
```

Comment:
`forEach` (the loops are only doing side effects)

Reply:
That's done 👍